### PR TITLE
upgrade hash op to support Tensor and LoDTensor input 

### LINKED ISSUE #17997: support Tensor input for hash op
upgrade hash op to support both Tensor and LoDTensor input

### PR DESCRIPTION
**Pull request by zhoukunsheng**
fix #17997

### CODE REVIEW

**Comment by zhoukunsheng** [comment]
image

**Review by XiaoguangHu01 — DISMISSED** [review]

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -10666,40 +10663,30 @@ def hash(input, hash_size, num_hash=1, name=None):
...
         name (str, default None): The name of this layer.
 
     Returns:
-       Variable: The hash result variable which is a LoDTensor.
+       Variable: The hash result variable, which the same variable type as `input`.
 
     Examples:
        .. code-block:: python
 
-            import paddle.fluid as fluid
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: Should keep these imports to make the example executable

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -10666,40 +10663,30 @@ def hash(input, hash_size, num_hash=1, name=None):
...
-            # Init Tensor
-            tensor = fluid.core.LoDTensor() 
-            tensor.set(np.random.randint(0, 10, (3, 1)).astype("int32"), place)
-            # Set LoD
-            tensor.set_recursive_sequence_lengths([[1, 1, 1]])
-
-            out = exece.run(feed={'titles': tensor}, fetch_list=[hash_r], return_numpy=False)
+            # titles has shape [batch, 1] and lod information
+            titles = fluid.layers.data(name='titles', shape=[1], dtype='int32', lod_level=1)
+            # hash_r has shape [batch, 2] and inherit lod information from titles
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: `inherit` -> `inherits`

Comment (zhoukunsheng) [reply python/paddle/fluid/layers/nn.py]: ok, thanks.

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (zhoukunsheng) [reply python/paddle/fluid/layers/nn.py]: ok, thanks.

**Comment by zhoukunsheng** [comment]
image

**Review by xsrobin — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]

**Pull request merged**